polkadot/node/core/backing/src/lib.rs

```diff
@@ -322,6 +322,8 @@ struct State {
 	/// This is guaranteed to have an entry for each candidate with a relay parent in the implicit
 	/// or explicit view for which a `Seconded` statement has been successfully imported.
 	per_candidate: HashMap<CandidateHash, PerCandidateState>,
+	/// Cache the per-session validators.
+	validators_cache: LruMap<SessionIndex, Vec<ValidatorId>>,
```

Comment:
We could map to an `Arc` here to avoid deep-cloning the vector of validator keys

Reply:
Let's also unify the per-session caches in a struct for convenience and provide a function for retrieving/updating